axon-framework/tuning/event-processing.md

```diff
@@ -0,0 +1,82 @@
+# Event Processing
+
+Typically, application components contain one or more[ Event Processors](../events/event-processors.md) which are responsible for processing incoming events. Tracking Event Processors have configuration aspects that can be changed at runtime to accommodate for changes in the system topology.‌
+
+## Increasing and decreasing segment counts <a id="increasing-and-decreasing-segment-counts"></a>
+
+Tracking Event Processors that handle events in multiple threads use segments to separate the events in the stream across these threads in a reliable way. However, especially when these threads are spread across multiple instances of a component, and the number of instances changes, it may be useful to scale the number of segments accordingly.‌
+
+To this end, Axon Framework provides a [split and merge API](../events/event-processors.md#splitting-and-merging-tracking-tokens). This API can be utilized directly via a client configuration or through Axon Server, where the latter takes required coordination into account.‌
+
+### Segment tuning through Axon Framework <a id="segment-tuning-through-axon-framework"></a>
+
+The Tracking Event Processors in Axon Framework provide methods to increase or decrease the number of segments for that particular instance. When using this API, one must provide the ID of the segment to increase/decrease. Additionally, the instance on which the method is invoked, must be actively processing that segment.‌
+
+First, the instance of the Tracking Event Processor must be obtained. This can be done using Axon's Configuration API like so:
+
+```java
+// The `Configuration` was returned through the `Configurer` or is available as a bean in the Spring Application Context
+public TrackingEventProcessor retrieveTrackingProcessor(org.axonframework.config.Configuration axonConfig,
+                                                        String processorName) {
+    return axonConfig.eventProcessingConfiguration()
+                     .eventProcessor(processorName) // This call returns an Optional
```

Comment:
Nit: The `EventProcessingConfiguration` now also provide a `EventProcessingConfiguration#eventProcessor(String, Class<T>)` method. Thus the `filter(eventProcessor -> eventProcessor instanceof TrackingEventProcessor)` line can be removed in favor of doing `.eventProcessor(processorName, TrackingEventProcessor.class)`